akka-actor/src/main/scala/akka/pattern/CircuitBreaker.scala

```diff
@@ -52,8 +58,17 @@ object CircuitBreaker {
    * @param callTimeout [[scala.concurrent.duration.FiniteDuration]] of time after which to consider a call a failure
    * @param resetTimeout [[scala.concurrent.duration.FiniteDuration]] of time after which to attempt to close the circuit
    */
-  def create(scheduler: Scheduler, maxFailures: Int, callTimeout: FiniteDuration, resetTimeout: FiniteDuration): CircuitBreaker =
+  def create(
+    scheduler:    Scheduler,
+    maxFailures:  Int,
+    callTimeout:  FiniteDuration,
+    resetTimeout: FiniteDuration): CircuitBreaker =
     apply(scheduler, maxFailures, callTimeout, resetTimeout)
+
+  def exceptionAsFailure[T]: Try[T] ⇒ Boolean = {
```

Comment:
Great question, do you a BiFunction would cut it?

I ain't familiar with Java, but seems CompletionStage can be handle by a Bifunction according to [docs](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletionStage.html#handle-java.util.function.BiFunction-)

We can allow user to pass a BiFunction which return boolean